site/content/api/Command-And-Control-API.md

```diff
@@ -12,43 +12,39 @@ THIS IS A FIRST DRAFT VERSION OF THE API SPEC AND THUS STILL SUBJECT TO DISCUSSI
-
-## Receive Control Message
+## Send and Receive a Command
 
 **Preconditions**
 
-1. Client has established an AMQP connection with Hono.
-2. Client has established an AMQP link in role *receiver* with Hono using target address `control/${tenant_id}/${device_id}`. Where `${device_id}` is the ID of the device to receive control messages for and `${tenant_id}` is the ID of the tenant the device belongs to.
+1. *Protocol Adapter* and *Business Application* have established an AMQP connection with the AMQP 1.0 Network.
+2. *Protocol Adapter* has established an AMQP link in role *receiver* with the AMQP 1.0 Network using target address `control/${tenant_id}/${device_id}`. Where `${device_id}` is the ID of the device to receive control messages for and `${tenant_id}` is the ID of the tenant the device belongs to. 
+3. *Business Application* has established an AMQP link in role *sender* (also using target address `control/${tenant_id}/${device_id}`). 
+
+**Link establishment and Message Flow**
 
-**Message Flow**
+The following sequence diagram shows the whole message exchange, together with the reply from the *Protocol Adapter* to the *Business Application* (which is defined above in detail).
 
-*TODO* add sequence diagram
+![Send Command](../command_control_Success.png)
 
 **Message Format**
 
-The following table provides an overview of the properties a client needs to set on a *Control* message.
+The following table provides an overview of the properties the *Business Application* needs to set on a *Control* message.
 
 | Name | Mandatory | Location | Type      | Description |
 | :--- | :-------: | :------- | :-------- | :---------- |
-| *message-id* | yes | *properties* | UTF-8 *string* | MUST contain the message ID used to match incoming control message with the response message containing result of execution. |
-| *device_id* | yes | *application-properties* | UTF-8 *string* | MUST contain the ID of the device the data in the payload has been reported by. |
-| *tenant_id* | yes | *application-properties* | UTF-8 *string* | MAY be set by the client to indicate which tenant the the device belongs to. |
-| *command* | no | *application-properties* | UTF-8 *string* | MUST contain the command to be executed by a device. |
-|*ttl* | no | *properties* | *int* | Indicates the timeout value (in milliseconds) indicating when server can assume that command has not been executed because of the lack of the response from the client. Default timeout is *60000* (1 minute). |
-|*durable* | no | *properties* | *boolean* | Indicates if the message should be persisted (*store and forward*) by the underlying message transport middleware. By default command&control messages are durable.  |
+| *message-id* | yes | *properties* | *string* | MUST contain the message ID used to match incoming control message with the response message containing result of execution. |
+| *subject* | yes | *properties* | *string* | MUST contain the command to be executed by a device. |
```

Comment:
What is the purpose of this property? Do we map this to a transport protocol specific property in the protocol adapters?

Reply:
It is the command, that must be known by the device. It could work for itself or indicates what to do with the data of the command. How me map this to the device protocols is out of scope for this API, IMPOV. In MQTT this could be the last part of the topic.

Reply:
My question was more if we make sure that this information is conveyed to the device so that it *can* make use of it.